### PR TITLE
daml script runner fails sooner if provided ledger host lacks a protocol when using the JSON API

### PR DESCRIPTION
**Pull request by moisesackerman-da**
Fixes https://github.com/digital-asset/daml/issues/19174

### CODE REVIEW

**Review by garyverhaegen-da — APPROVED** [review]
Thank you!

**Comment by garyverhaegen-da** [comment]
Would this be worth backporting to 2.8?

**Comment by moisesackerman-da** [comment]
> Would this be worth backporting to 2.8?

yes I think so, perhaps even 2.7.x and 2.3.x too; the use of `Uri(_)` goes way back - see https://github.com/digital-asset/daml/pull/9057, first released with Daml SDK 1.12.0, which clarified a few of the `daml script` options, mentions that `--ledger-host` requires a protocol if using the JSON API. Ofc if the presumably few users of the older SDKs haven't complained so far it's unlikely they will complain at all, so I'll leave the backporting to you

File: sdk/daml-script/runner/src/test/resources/TestScript.daml [hunk 1]
```diff
@@ -5,18 +5,18 @@ module TestScript where
 
 import Daml.Script
 
--- Each script gets the time to ensure the ledger is called
+-- Each script lists the known parties to ensure the ledger is called
 myScript : Script ()
 myScript = do
-  _ <- getTime
```
Comment (moisesackerman-da) [inline sdk/daml-script/runner/src/test/resources/TestScript.daml:11]: these tests would have caught the missing protocol errors if it hadn't been for the fact that the default time mode is `--wall-clock-time`, which does _not_ call to the ledger :(

**Pull request merged**